src/database/mod.rs

```diff
@@ -840,6 +833,135 @@ pub trait DatabaseExt<'a>: Executor<'a, Database = Postgres> {
         let row_unprocessed = self.fetch_one(query_queued_deletion).await?;
         Ok(row_unprocessed.get::<bool, _>(0))
     }
+
+    async fn insert_new_batch_head(
+        self,
+        next_root: &Hash,
+        batch_type: BatchType,
+        commitments: &Commitments,
+        leaf_indexes: &LeafIndexes,
+    ) -> Result<(), Error> {
+        let query = sqlx::query(
+            r#"
+            INSERT INTO batches(
+                next_root,
+                prev_root,
```

Comment:
we can omit prev_root here, no?

Reply:
I'm also fine with explicit null

Reply:
I personally prefer to be more explicit (not to check schema on table for every insert) but I'm fine with both approaches.

Reply:
Ok, let's leave it like it is